service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionService.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.fpl.service.casesubmission;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisCaseSubmission;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C110A;
+import static uk.gov.hmcts.reform.fpl.utils.SubmittedFormFilenameHelper.buildFileName;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseSubmissionService  {
+    private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+    private final CaseSubmissionTemplateDataGenerationService documentGenerationService;
+    private final ObjectMapper objectMapper;
+
+    public Document generateSubmittedFormPDF(final CaseDetails caseDetails) {
+        CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+        DocmosisCaseSubmission submittedCase = documentGenerationService.getTemplateData(caseData);
+
+        DocmosisCaseSubmission submittedCaseWithCaseNumber =
+            populateDocmosisCaseSubmissionWithCaseNumber(submittedCase, caseDetails.getId());
+
+        DocmosisDocument document = docmosisDocumentGeneratorService.generateDocmosisDocument(
+            submittedCaseWithCaseNumber, C110A);
+
+        return uploadDocumentService.uploadPDF(document.getBytes(), buildFileName(caseDetails));
+    }
+
+    private DocmosisCaseSubmission populateDocmosisCaseSubmissionWithCaseNumber(
+        final DocmosisCaseSubmission submittedCase,
+        final long caseNumber) {
+        return submittedCase.toBuilder().caseNumber(String.valueOf(caseNumber)).build();
+    }
```

Comment:
why is all this casenumber stuff separate to the template generation method?

Reply:
CaseNumber is in caseDetails not in caseData. But as part of abstract method getTemplateData can accept only caseData. So, adding caseNumber here to the builder

Reply:
Maybe if we didn't have the all the variables as final then we wouldn't need to clone the object to add the required field.

Reply:
Moved this casenumber stuff to templateGenerationService.